### DISCUSSION TITLE
how do I check if a player is falling?

### DISCUSSION THREAD

**Discussion by hcnery08alt** [Minestom/Minestom#1901] [Help]
player.getVelocity() gets the velocity applied onto the player I think not the current velocity

 here is my code:

```java
    public static boolean isCritical(LivingEntity target) {

        boolean critical = target.getVelocity().y() < 0
                && !target.isOnGround()
                && !target.isSprinting();

        // testing
        ((Player) target).sendMessage(critical + " at " + target.getVelocity().y());

        return critical;
    }
```

**Comment by DevBlokJochem** [comment]
You check if the player is not on the ground. I believe that there is a function called Player#onGround()

**Reply by hcnery08alt** [reply]
> You check if the player is not on the ground. I believe that there is a function called Player#onGround()

yes but this woukd retern true if the player was jumping. Is there a way to get the player velocity relative to the world so if the y velocity is small than 0 the player will be traverling dowwards

**Comment by KrystilizeNevaDies** [comment]
You should be able to do something like `player.getVelocity().y() < 0`

**Reply by hcnery08alt** [reply]
It doesn't work. Did you read my code?
